optuna/samplers/_tpe/sampler.py

```diff
@@ -513,13 +514,16 @@ def _get_internal_repr(
         return {k: np.asarray(v) for k, v in values.items()}
 
     def _sample(
-        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        search_space: dict[str, BaseDistribution],
+        use_cache: bool,
```

Comment:
[nit] A more informative argument name would be preferred. 

```suggestion
        use_trial_cache: bool,
```